aiida/engine/processes/process.py

```diff
@@ -103,12 +103,11 @@ class SaveKeys(enum.Enum):
     @classmethod
     def define(cls, spec):
         super(Process, cls).define(spec)
-        spec.input_namespace(spec.metadata_key, required=False, non_db=True, default={})
+        spec.input_namespace(spec.metadata_key, required=False, non_db=True, valid_type=dict, default={})
```

Comment:
As a note here, also `default={}` is not the right thing to do, because the *same* dictionary will be shared by all instances that use the default.

Reply:
You are right, but actually the `valid_type` I added is also wrong. I was intending to say that the value passed to this namespace should be a `dict`, but what it really is saying is that _each_ entry in the namespace should be of type `dict` which is wrong.

Reply:
The `default` was put there because without it, everything would break :+1:  but I have addressed the underlying issue in `plumpy` and made a new release `0.12.2` which is now included in this fix